asistente_ladm_col/logic/quality_rules/qr_building_duplicate_records.py

```diff
@@ -0,0 +1,97 @@
+        begin                : 2022-06-15
+        git sha              : :%H$
+        copyright            : (C) 2022 by Sergio Ramírez (BSF Swissphoto)
+        email                : seralra96@gmail.com
+ ***************************************************************************/
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License v3.0 as          *
+ *   published by the Free Software Foundation.                            *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from qgis.PyQt.QtCore import (QCoreApplication,
+                              QSettings)
+
+from asistente_ladm_col.config.enums import EnumQualityRuleResult
+from asistente_ladm_col.config.keys.common import QUALITY_RULE_LADM_COL_LAYERS
+from asistente_ladm_col.config.layer_config import LADMNames, LayerConfig
+from asistente_ladm_col.config.quality_rule_config import (QR_IGACR4016,
+                                                           QRE_IGACR4016E01)
+from asistente_ladm_col.core.quality_rules.abstract_logic_quality_rule import AbstractLogicQualityRule
+from asistente_ladm_col.core.quality_rules.quality_rule_execution_result import QualityRuleExecutionResult
+from asistente_ladm_col.logic.ladm_col.ladm_data import LADMData
+
+
+class QRDuplicateBuildingRecords(AbstractLogicQualityRule):
+    """
+    Check that building don't have duplicate records
+    """
+    _ERROR_01 = QRE_IGACR4016E01  # Building with duplicate records
+
+    def __init__(self):
+        AbstractLogicQualityRule.__init__(self)
+
+        self._id = QR_IGACR4016
+        self._name = "Revisar que la tabla construccion no tenga registros repetidos"
+        self._tags = ["igac", "instituto geográfico agustín codazzi", "lógica", "negocio", "construccion", "registros repetidos"]
+        self._models = [LADMNames.SURVEY_MODEL_KEY]
+
+        self._errors = {self._ERROR_01: "Construccion no debe tener registros repetidos"}
+
+        # Optional. Only useful for display purposes.
+        self._field_mapping = dict()  # E.g., {'id_objetos': 'ids_punto_lindero', 'valores': 'conteo'}
+
+    def layers_config(self, names):
+        return {QUALITY_RULE_LADM_COL_LAYERS: [names.LC_BUILDING_T]}
+
+    def _validate(self, db, db_qr, layer_dict, tolerance, **kwargs):
+        self.progress_changed.emit(5)
+        ladm_queries = self._get_ladm_queries(db.engine)
+
+        pre_res, pre_obj = self._check_prerrequisite_layers(layer_dict)
+        if not pre_res:
+            return pre_obj
+
+        error_state = None
+        
+        fields = LayerConfig.get_logic_consistency_tables(db.names).get(db.names.LC_BUILDING_T)
```

Comment:
I propose to eliminate the `LayerConfig.get_logic_consistency_tables()` method and encapsulate the logic in each of the quality rules.